tls/s2n_connection.c

```diff
@@ -1537,3 +1538,19 @@ int s2n_connection_get_selected_client_cert_signature_algorithm(struct s2n_conne
 
     return S2N_SUCCESS;
 }
+
+/*
+ * Gets the config set on the connection.
+ */
+int s2n_connection_get_config(struct s2n_connection *conn, struct s2n_config **config) {
+    POSIX_ENSURE_REF(conn);
+    POSIX_ENSURE_REF(config);
+
+    if (s2n_fetch_default_config() == conn->config) {
+        POSIX_BAIL(S2N_ERR_NULL);
+    }
```

Comment:
Just to make sure this is intentional: the behavior you want is to return an error if the config hasn't been set?

Reply:
I flopped on if its really an error. But yea I think this way will be easier to consume since we can couple it with the `ERR_NULL`. If we return Success then we would need another way to communicate.. no its actually not present and here is a NULL pointer.